### ISSUE TITLE
Add note to docs: Source Maps are not uploaded for dev/debug builds

### ISSUE THREAD

**Issue by reizhafajrian** [getsentry/sentry-react-native#3743]
**OS:**
- [ ]  Windows
- [x]  MacOS
- [ ]  Linux

**Platform:**
- [x]  iOS
- [ ]  Android

**SDK:**
- [x]  `@sentry/react-native` (>= 1.0.0)
- [ ]  `react-native-sentry` (<= 0.43.2)

**SDK version:** 0.0.0

**`react-native` version:** 0.0.0

**Are you using Expo?**
- [x]  Yes
- [ ]  No

**Are you using sentry.io or on-premise?**
- [x]  sentry.io (SaaS)
- [ ]  on-premise

**If you are using sentry.io, please post a link to your issue so we can take a look:**

[Link to issue]

**Configuration:**

(`app.json`)
```
"plugins": [
      [
        "@sentry/react-native/expo",
        {
          "url": "https://sentry.io/",
          "project": "example-5",
          "organization": "example"
        }
      ]
    ],
```

(`@sentry/react-native`)
```
Sentry.init({
  dsn: "mydsn",
  tracesSampleRate: 1.0,
});

//app.js code
export default Sentry.wrap(App);
```

(`metro.config.js`)
```
const { getSentryExpoConfig } = require('@sentry/react-native/metro');

const config = getSentryExpoConfig(__dirname);

module.exports = config;
```

---
**I have the following issue:**

[Description]

**Steps to reproduce:**
- following setup like sentry docs with expo
- build the app with eas

**Actual result:**

<img width="1208" alt="image" src="https://github.com/getsentry/sentry-react-native/assets/65649495/a47a516d-176b-46c2-86ee-ca2ad0d7fd5c">

**Expected result:**

Source map should be there

**Comment by reizhafajrian** [comment]
I recently discovered that in Expo 50, source maps are not uploaded when using debug or simulator versions. This detail wasn't immediately clear in the current documentation, and I only realized it after manually checking the repository.

For the benefit of other developers, could the documentation be updated to clearly state that source maps do not get uploaded in debug or simulator modes? Adding a specific note or warning about this in the relevant sections could prevent future confusion and help developers better understand the environment-specific behaviors of Expo.

Thank you for your attention to this matter
